### PR TITLE
switch to nauty for generating posets

### PR DESCRIPTION
**Pull request by fchapoton**
as there is now a direct method in `nauty` and we have added the interface previously

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

**Review by dcoudert — APPROVED** [review]
LGTM.

File: src/sage/combinat/posets/posets.py [hunk 1]
```diff
@@ -8945,8 +8945,8 @@ def __iter__(self):
             sage: list(P)
             [Finite poset containing 2 elements, Finite poset containing 2 elements]
         """
-        from sage.graphs.digraph_generators import DiGraphGenerators
-        for dig in DiGraphGenerators()(self._n, is_poset):
+        from sage.graphs.digraph_generators import DiGraphGenerators as DG
+        for dig in DG().nauty_posetg(f"{self._n} o"):
```
Comment (mantepse) [inline src/sage/combinat/posets/posets.py]: I would prefer not to use `as DG`, because it makes code much harder to read for me personally.  Also, the import could actually go to the module level, no?

File: src/sage/combinat/tutorial.py [hunk 2]
```diff
@@ -844,15 +844,15 @@
 
 Partial orders on a set of `8` elements, up to isomorphism::
 
-    sage: C = Posets(8); C
-    Posets containing 8 elements
+    sage: C = Posets(7); C
+    Posets containing 7 elements
     sage: C.cardinality()
-    16999
+    2045
```
Comment (mantepse) [inline src/sage/combinat/tutorial.py:850]: Why this change?  On my computer, even *generating* all the posets now takes only half a second.

**Comment by mantepse** [comment]
This is a very impressive speedup:
```
sage: %time n=7; sum(1 for dig in DiGraphGenerators()(n, is_poset))
CPU times: user 5.85 s, sys: 0 ns, total: 5.85 s
Wall time: 5.85 s
2045
sage: %time n=7; sum(1 for dig in DiGraphGenerators().nauty_posetg(f"{n} o"))
CPU times: user 57.3 ms, sys: 0 ns, total: 57.3 ms
Wall time: 66.7 ms
2045
sage: %time n=8; sum(1 for dig in DiGraphGenerators().nauty_posetg(f"{n} o"))
CPU times: user 395 ms, sys: 0 ns, total: 395 ms
Wall time: 399 ms
16999
```

**Comment by dcoudert** [comment]
Actually, there is another issue: `nauty_posetg` works only for $n\leq 16$. You could use a switch to use it only in this case and use the previous (slower) method otherwise.

File: src/sage/combinat/tutorial.py (hunk 2 above)
Comment (fchapoton) [reply src/sage/combinat/tutorial.py:850]: well, I think smaller doctests are always better. No need to show muscles in doctests.

**Comment by mantepse** [comment]
> Actually, there is another issue: `nauty_posetg` works only for n ≤ 16 . You could use a switch to use it only in this case and use the previous (slower) method otherwise.

Great catch!

To avoid such oversights in future (not in this pull request!), wouldn't it be better to have intermediate methods that call `nauty_posetg` and do the argument parsing?  I am frequently wasting time looking up the precise string for `nauty_geng` to generate connected graphs only :-)

File: src/sage/combinat/tutorial.py (hunk 2 above)
Comment (mantepse) [reply src/sage/combinat/tutorial.py:850]: OK, I don't mind - although I somewhat disagree, with two (implicit) statements:

* in the case at hand, we are only looking up the number
* in general, I think it it good to use the largest example which takes a reasonable amount of time (in particular, less than a second), because it seems to me that this helps to catch bad performance regressions.

**Comment by mantepse** [comment]
I think that the issue pointed out by @dcoudert still needs fixing.

**Comment by fchapoton** [comment]
We lose a lot of speed in the relabeling..

**Comment by fchapoton** [comment]
merci pour les suggestions

**Review by dcoudert — APPROVED** [review]
LGTM.
